docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,491 @@
+     - True, False
+     - True      
+   * - ENABLE_PORTBLAS_BACKEND
+     - True, False
+     - False      
+   * - ENABLE_PORTFFT_BACKEND
+     - True, False
+     - False      
+   * - BUILD_FUNCTIONAL_TESTS
+     - True, False
+     - True      
+   * - BUILD_EXAMPLES
+     - True, False
+     - True      
+   * - TARGET_DOMAINS (list)
+     - blas, lapack, rng, dft, sparse_blas
+     - All domains 
+
+Some additional build options are given in the section `Additional build options`_.
+
+.. _build_target_domains:
+
+TARGET_DOMAINS
+^^^^^^^^^^^^^^
+
+OneMKL supports multiple domains: BLAS, DFT, LAPACK, RNG and sparse BLAS.
+The domains built by oneMKL can be selected using the ``TARGET_DOMAINS`` parameter.
+In most cases, ``TARGET_DOMAINS`` is set automatically according to the domains supported
+by the backend libraries enabled.
+However, while most backend libraries support only one of these domains, but some may support multiple.
+For example, the ``MKLCPU`` backend supports every domain. To enable support for only the BLAS domain in
+the oneMKL interface library whilst compiling with ``MKLCPU``, ``TARGET_DOMAINS`` could be set to ``blas``. 
+To enable BLAS and DFT, ``-DTARGET_DOMAINS="blas dft"`` would be used.
+
+
+.. _build_for_intel_onemkl_dpcpp:
+
+Building for Intel(R) oneMKL
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The closed-source Intel(R) oneMKL supports multiple domains on both x86 CPUs and Intel GPUs.
+The MKLCPU backend using Intel(R) oneMKL for x86 CPU is enabled by default, and controlled with the parameter ``ENABLE_MKLCPU_BACKEND``.
+The MKLGPU backend using Intel(R) oneMKL for Intel GPU is enabled by default, and controlled with the parameter ``ENABLE_MKLGPU_BACKEND``.
+
+When using the Intel(R) oneAPI DPC++ Compiler, it is likely that Intel(R) oneMKL will be found automatically.
+If it is not, the parameter ``MKL_ROOT`` can be set to point to the installation prefix of Intel(R) oneMKL.
+
+
+.. _build_for_CUDA_dpcpp:
+
+Building for CUDA
+~~~~~~~~~~~~~~~~~
+
+The CUDA backends can be enabled with ``ENABLE_CUBLAS_BACKEND``, ``ENABLE_CUFFT_BACKEND``, ``ENABLE_CURAND_BACKEND``,
+and ``ENABLE_CUSOLVER_BACKEND``.
+
+No additional parameters are required for using CUDA libraries. In most cases, the CUDA libraries should be
+found automatically by CMake.
+
+.. _build_for_ROCM_dpcpp:
+
+Building for ROCm
+~~~~~~~~~~~~~~~~~
+
+The ROCm backends can be enabled with ``ENABLE_ROCBLAS_BACKEND``, ``ENABLE_ROCFFT_BACKEND``, ``ENABLE_ROCSOLVER_BACKEND`` and ``ENABLE_ROCRAND_BACKEND``.
+
+For *RocBLAS*, *RocSOLVER* and *RocRAND*, the target device architecture must be set. This can be set with using the ``HIP_TARGETS`` parameter.
+For example, to enable a build for MI200 series GPUs, ``-DHIP_TARGETS=gfx90a`` should be set.
+Currently, DPC++ only supports a single HIP target. This may change in future versions.
+
+A few often-used architectures are listed below:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Architecture
```

Comment:
This table should be copied in (or linked from) the AdaptiveCpp document.

Reply:
In the AdaptiveCpp document I've said:
```quote
For common AMD GPU architectures, see the :ref:`build_for_ROCM_dpcpp` in the DPC++ build guide.
```